python/ray/data/aggregate.py

```diff
@@ -217,6 +220,188 @@ def finalize(a: List[float]):
         )
 
 
+@PublicAPI
+class VectorizedAggregation:
+    """Defines a vectorized aggregation for tabular datasets, where both per-block
+    aggregations and cross-block aggregation merges are vectorized.
+    """
+
+    def __init__(
+        self,
+        aggregate_block: Callable[[TableBlockAccessor, str], U],
```

Comment:
This could be generalized to groupby by calling this on a slice of the table block instead of the whole, right?

I.e., we could probably use this current code for groupby case too.